proposals/NNNN-Processing-of-unknown-enum-values-by-SDL-and-Mobile-libraries.md

```diff
@@ -0,0 +1,176 @@
+      
+      1.2 Parameter is optional: `audioStreamingIndicator="UNKNOWN"`
+      SDL core removes the `UNKNOWN` item from the request.
+      The value of `audioStreamingIndicator` parameter becomes empty.
+      SDL core omits the empty parameter in the request.
+      Since the parameter is optional (according to Mobile API), the request is successfully processed with the following response to Mobile Application:
+      `success=true, resultCode = "WARNINGS", info="Invalid enums were removed: audioStreamingIndicator:UNKNOWN"`
+      
+  2. Request contains a parameter which value is an array of enum items
+      Request: `RegisterAppInterface`
+      
+      2.1 Only one enum item is unknown: `appHMIType = [ "DEFAULT", "UNKNOWN"]`
+      SDL core removes the `UNKNOWN` item from the request.
+      The value of `appHMIType` parameter now contains only `DEFAULT` item (which is known one).
+      Request is successfully processed with the following response to Mobile Application:
+      `success=true, resultCode = "WARNINGS", info="Invalid enums were removed: appHMIType:UNKNOWN"`
+      
+      2.2 All enum items are unknown: `appHMIType = [ "UNKNOWN_1", "UNKNOWN_2"]`
+      SDL  core removes both `UNKNOWN_1` and `UNKNOWN_2` items from the request.
+      The value of `appHMIType` parameter becomes empty.
+      Since `minsize` of the parameter is `1` (according to Mobile API), the request fails with the following response to Mobile Application:
+      `success=false, resultCode = "INVALID_DATA", info="Invalid enums were removed: appHMIType:UNKNOWN_1, UNKNOWN_2"`
+      
+  3. Request contains a parameter (with enum item value) which is a part of the structure
+      Request: `SetGlobalProperties`
+      Parameter `menuIcon="Image"`, sub-parameter: `imageType`, `type="ImageType"`
+      
+      ```
+      menuIcon = {
+        imageType = "UNKNOWN"
+      }
+      ```
+      
+      SDL core removes the `menuIcon` Parameter from the request since the parameter `imageType` is mandatory for the `ImageType` structure.
+      However, request will be processed successfully since the `menuIcon` parameter is optional for `SetGlobalProperties` request and the following response will be provided to Mobile Application:
+      
+      `success=true, resultCode = "WARNINGS", info="Invalid enums were removed: menuIcon.imageType:UNKNOWN"`
+      
+  4. Request contains a parameter which is an array of elements and each element has a parameter which value is an enum item.
+      Request: `Show`
+      Parameter: `softButtons`, `type="SoftButton"`, sub-parameter: `type`, `type="SoftButtonType"`
+      There are 2 soft buttons provided in the request:
+      
+      ```
+      softButtons = [
+        { softButtonID = 1, type = "UNKNOWN" },
+        { softButtonID = 2, type = "IMAGE" }
+      ]
+      ```
+      
+      SDL core removes the whole element with the button `1` from the request since the `type` parameter is mandatory for `SoftButton` structure:
```

Comment:
```suggestion
      SDL Core removes the whole element with the button `1` from the request since the `type` parameter is mandatory for `SoftButton` structure:
```